src/Logging/Logging.CodeGenerator/LogScopeGenerator.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Logging.CodeGenerator
+{
+    public static class LogScopeGenerator
+    {
+        private static readonly string EndOfLine = @"
+";
+
+
+        public static IEnumerable<(string fileName, string fileContent)> Generate()
+        {
+            var infos = Enumerable.Range(0, 3).Select(BuildInfo);
```

Comment:
I suppose makes sense for LogMessage to allow more generic parameters than LogScope, but I think a max of 2 for LogScope is pretty small. What do people thing about allowing up to 4 generic parameters?

Reply:
Oops! That should have been `Range(1, 3)` - the methods to define delegates only goes up to `LoggerMessage.DefineScope<T1, T2, T3>`... Should this match T3 or should we add T4 or more to LoggerMessage.DefineScope?